modules/core/src/lib/layer.ts

```diff
@@ -1066,9 +1067,9 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
     // @ts-ignore (TS2339) internalState is alwasy defined when this method is called
     this.props = this.internalState.propsInTransition || currentProps;
 
-    const opacity = this.props.opacity;
     // apply gamma to opacity to make it visually "linear"
-    uniforms.opacity = Math.pow(opacity, 1 / 2.2);
+    const opacity = Math.pow(this.props.opacity, 1 / 2.2);
+    uniforms.opacity = opacity; // TODO remove once layers ported to UBO
```

Comment:
Unrelated to your PR, but FYI:

- https://github.com/visgl/deck.gl/issues/8972